src/ucp/rma/basic_rma.c

```diff
@@ -113,15 +134,39 @@ static ucs_status_t ucp_progress_put_nbi(uct_pending_req_t *self)
                                       req->send.rma.remote_addr,
                                       uct_rkey);
         status = (packed_len > 0) ? UCS_OK : (ucs_status_t)packed_len;
+    } else {
+        uct_iov_t iov;
+        packed_len = req->send.length;
+        // TODO thresholds
+        iov.buffer = (void *)req->send.buffer;
+        iov.length = packed_len;
+        iov.count  = 1;
+        iov.memh   = req->send.state.dt.contig.memh;
+        ++req->send.uct_comp.count;
+
+        status = uct_ep_put_zcopy(ep->uct_eps[req->send.lane], 
+                                  &iov, 1, 
+                                  req->send.rma.remote_addr,
+                                  uct_rkey,
+                                  &req->send.uct_comp);
+        if (ucs_unlikely(status <= 0)) {
```

Comment:
actually it depends on the uct transport. Ugni and ib always return in progress while mm*s return ok